pkg/types/vsphere/platform.go

```diff
@@ -22,7 +22,7 @@ type Platform struct {
 	Folder string `json:"folder,omitempty"`
 
 	// Cluster is the name of the cluster virtual machines will be cloned into.
-	Cluster string `json:"cluster,omitempty"`
+	Cluster string `json:"cluster"`
```

Comment:
@patrickdillon UPI today doesn't require `Cluster` because terraform creates a resource pool.  In the IPI case we use the root resource pool - which requires the cluster name to retrieve the resource pool id.

This change will break the UPI CI template.  It would be a very minor fix.

Reply:
I have dropped the commit to require cluster. We need to find a solution but let's roll it into a different PR, as this one should otherwise be ready.